src/main/scala/com/softwaremill/realworld/users/UsersEndpoints.scala

```diff
@@ -0,0 +1,42 @@
+package com.softwaremill.realworld.users
+
+import com.softwaremill.realworld.auth.AuthService
+import com.softwaremill.realworld.db.{Db, DbConfig}
+import com.softwaremill.realworld.utils.*
+import io.getquill.SnakeCase
+import sttp.model.StatusCode
+import sttp.tapir.generic.auto.*
+import sttp.tapir.json.zio.jsonBody
+import sttp.tapir.server.ServerEndpoint.Full
+import sttp.tapir.ztapir.*
+import sttp.tapir.{EndpointInput, PublicEndpoint, Validator}
+import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}
+import zio.{Cause, Exit, ZIO, ZLayer}
+
+import javax.sql.DataSource
+
+class UsersEndpoints(usersService: UsersService, base: BaseEndpoints):
+
+  import UsersEndpoints.given
+
+  val get: ZServerEndpoint[Any, Any] = base.secureEndpoint.get
+    .in("api" / "user")
+    .in(header[String]("Authorization"))
```

Comment:
this is already included in the `secureEndpoint`, along with handling authentication

Reply:
yes, indeed, however in this case we still use this header to determine user identity in logic of [get current user](https://realworld-docs.netlify.app/docs/specs/backend-specs/endpoints/#get-current-user) endpoint:
```
Get Current User
GET /api/user

Authentication required, returns a [User](https://realworld-docs.netlify.app/docs/specs/backend-specs/api-response-format#users-for-authentication) that's the current user
```

Reply:
but the security logic returns a `UserSession`, which you can use here to calculate the response in the server logic?

Reply:
thank you for the hint